include/clipp.h

```diff
@@ -160,7 +165,14 @@ constexpr auto
 check_is_callable(int) -> decltype(
     std::declval<Fn>()(std::declval<Args>()...),
     std::integral_constant<bool,
-        std::is_same<Ret,typename std::result_of<Fn(Args...)>::type>::value>{} );
+        std::is_same<Ret,
+#if CLIPP_CPLUSPLUS >= 201703L
+        typename std::invoke_result<Fn(Args...)>::type
```

Comment:
Using Apple Clang version 14.0.3 (C++17, C++20), this will not compile.
Shouldn't it be `typename std::invoke_result<Fn, Args...>` ?
One can also use the shorter version `std::invoke_result_t<Fn, Args...>`.

Reply:
Fix it.